src/utils/db_manager.py

```
@@ -1,49 +1,170 @@
+        Return the set of unique Email values stored in the scored_leads table.
+        
+        Returns an empty set if the query fails (errors are logged). The database
+        connection is always closed before returning.
+        """
+        try:
+            conn = sqlite3.connect(self.db_path)
+            # Use pandas for efficient querying
+            df = pd.read_sql_query("SELECT Email FROM scored_leads", conn)
+            return set(df['Email'].unique())
+        except (sqlite3.Error, pd.errors.DatabaseError) as e:
+            logging.error(f"Could not fetch existing emails from database: {e}")
```

Ruff (0.12.2)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)